src/scripts/main.js

```diff
@@ -1,3 +1,254 @@
+        ? element.value
+        : parseInt(element.value);
+    }
+
+    return data;
+  }, {});
+
+  if (!validateForm(formData)) {
+    return;
+  }
+
+  pushNotification('success', 'Form submitted successfully');
+
+  addRowToTable(formData);
+});
+
+function parseCellValue(row, index) {
+  const cells = row.querySelectorAll('td');
+
+  if (!cells.length) {
+    return '';
+  }
+
+  const cleanedCell = cells[index].textContent.trim().replace(/[$,]/g, '');
+
+  return isNaN(cleanedCell) ? cleanedCell : parseInt(cleanedCell);
+}
+
+tableHeader.addEventListener('click', (e) => {
+  const clickedCell = e.target.closest('th');
+
+  if (!clickedCell) {
+    return;
+  }
+
+  const columnIndex = clickedCell.cellIndex;
+  const tableRows = [...document.querySelectorAll('tbody tr')];
+  const tbody = document.querySelector('tbody');
+
+  if (lastSortedColumnIndex !== columnIndex) {
+    tableRows.sort((row1, row2) => {
+      const value1 = parseCellValue(row1, columnIndex);
+      const value2 = parseCellValue(row2, columnIndex);
+
+      if (typeof value1 === 'number' && typeof value2 === 'number') {
+        return value1 - value2;
+      }
+
+      if (typeof value1 === 'string' && typeof value2 === 'string') {
+        return value1.localeCompare(value2);
+      }
+    });
+
+    lastSortedColumnIndex = columnIndex;
+  } else {
+    tableRows.reverse();
+  }
+
+  tableRows.forEach((rowData) => {
+    tbody.appendChild(rowData);
+  });
+});
+
+tableBody.addEventListener('click', (e) => {
+  const clickedRow = e.target.closest('tr');
+
+  if (!clickedRow || activeTableRow === clickedRow) {
+    return;
+  }
+
+  if (activeTableRow) {
+    activeTableRow.classList.remove('active');
+  }
+
+  activeTableRow = clickedRow;
+  activeTableRow.classList.add('active');
+});
+
+function pushNotification(type, message) {
+  const notificationBlock = document.createElement('div');
+  const title = document.createElement('h1');
+  const phrase = document.createElement('p');
+
+  notificationBlock.setAttribute('data-qa', 'notification');
+  notificationBlock.className = `notification ${type}`;
+  title.textContent = type[0].toUpperCase() + type.slice(1);
+  phrase.textContent = message[0].toUpperCase() + message.slice(1);
+
+  notificationBlock.appendChild(title);
+  notificationBlock.appendChild(phrase);
+  document.body.appendChild(notificationBlock);
+
+  setTimeout(() => {
+    notificationBlock.style.visibility = 'hidden';
+  }, NOTIFICATION_TIMEOUT);
+}
+
+function validateForm(formData) {
+  if (
+    !formData.name ||
+    !formData.position ||
+    !formData.age ||
+    !formData.salary
+  ) {
+    pushNotification('error', 'Each field must be filled in');
+
+    return false;
+  }
+
+  if (formData.name.length < 4) {
+    pushNotification('error', 'Name must be no shorter than 4');
+
+    return false;
+  }
+
+  if (formData.age < MIN_AGE || formData.age > MAX_AGE) {
+    pushNotification('error', 'Age must be between 18 and 90');
```

Comment:
The error message 'Age must be between 18 and 90' should reflect the constants MIN_AGE and MAX_AGE to ensure consistency. Consider using template literals to dynamically include these constants in the message.